client/file-browser/file-browser.tsx

```diff
@@ -600,59 +406,55 @@ export function FileBrowser({
     },
   })
 
-  const noRowsRenderer = useCallback(() => {
-    if (isLoadingFiles) {
-      return <LoadingDotsArea />
-    } else if (loadFilesError) {
-      return <ErrorText>{loadFilesError.message}</ErrorText>
-    } else {
-      return <EmptyText>Nothing to see here</EmptyText>
+  const renderRow = useStableCallback((index: number) => {
```

Comment:
Does react-virtuoso call this during render? If so, it won't be safe to use this function (we have no way of enforcing this, but React's implementation will)

Reply:
I added some logging where we do this in Ladder and it *seems* like the layout effect for the callback always runs before the function is called, so I think this is okay